ui/dashboards/src/components/Variables/Variable.tsx

```diff
@@ -37,42 +37,88 @@ export function TemplateVariable({ name }: TemplateVariableProps) {
 function ListVariable({ name }: TemplateVariableProps) {
   const ctx = useTemplateVariable(name);
   const definition = ctx.definition as ListVariableDefinition;
-  const { setVariableValue, loadTemplateVariable } = useTemplateVariableActions();
+  const variablePlugin = useVariablePlugin(definition);
+
+  const { setVariableValue, setVariableLoading, setVariableOptions } = useTemplateVariableActions();
   const allowMultiple = definition?.spec.allowMultiple === true;
+  const allowAllValue = definition?.spec.allowAllValue === true;
+
+  const loadOptions = useCallback(async () => {
+    if (!variablePlugin) {
+      return;
+    }
+    setVariableLoading(name, true);
+    try {
+      const { data } = await variablePlugin.getVariableOptions(definition);
+      setVariableOptions(name, data);
+    } catch (e) {
+      console.error('Failed to load template variable options', e);
+    }
+    setVariableLoading(name, false);
+  }, [variablePlugin, definition, name, setVariableLoading, setVariableOptions]);
+
   useEffect(() => {
-    loadTemplateVariable(name);
-  }, [name, loadTemplateVariable]);
+    loadOptions();
+  }, [loadOptions]);
```

Comment:
I wonder if there's some way to (eventually) move this logic up into the store? Or maybe make use of `useQueries` from `react-query` to run multiple options loading callbacks and manage the loading/data state? Anyway, just thinking out loud, nothing to do here.

Reply:
Yeah agreed, I tried to do this at first but it felt kind of awkward since I need to resolve the plugin through a hook . I was hoping that I could load it purely through async promises. That way you could load the options without mounting any components.

```ts
async loadVariable(name: string) {
   const def = getVariableDefinitionByName(name)
   const plugin = await loadPlugin(def.kind) 
}
```

The other idea was to load the plugin via a hook then set getVariableOptions function in the store. 
```tsx
store.setVariableLoader('myVar', plugin.getVariableOptions)

// Then later you can call it 
store.loadVariable('myVar')

```

Reply:
Yeah, the new `PluginRegistry` implementation I added should let you do that (part of the reason I redid the API was this use case). It's not in there quite yet, but once it is, you could probably use the context object directly:

https://github.com/perses/perses/blob/ec485059dbb767993ad071077c10c8fc3a32ce6f/ui/plugin-system/src/components/PluginRegistry/plugin-registry-model.ts#L17-L20